markets/bfp-market/contracts/modules/PerpAccountModule.sol

```diff
@@ -378,10 +343,25 @@ contract PerpAccountModule is IPerpAccountModule {
             market,
             toMarginValues.marginUsd.toInt() - toMarginValues.collateralUsd.toInt()
         );
+        // Stack to deep.
+        {
+            uint256 supportedSynthMarketIdsLength = globalMarginConfig
+                .supportedSynthMarketIds
+                .length;
+            uint128 currentSynthMarketId;
+            for (uint256 i = 0; i < supportedSynthMarketIdsLength; ) {
+                currentSynthMarketId = globalMarginConfig.supportedSynthMarketIds[i];
+                // Move collateral `from` -> `to`.
+                toAccountMargin.collaterals[currentSynthMarketId] += fromAccountMargin.collaterals[
```

Comment:
You should still check

```
fromAccountMargin.collaterals[currentSynthMarketId] != 0
```

So this loop wrt gas should be fairly equivalent to the removed `getMatchingMarketCollateral`.

Reply:
ops, will fix